crates/runtime/src/status.rs

```diff
@@ -139,6 +139,27 @@ impl RuntimeStatus {
         metrics::views::STATUS.record(status as u64, &[KeyValue::new("view", view_name)]);
     }
 
+    /// Update the status of a worker
+    pub fn update_worker(&self, name: &str, status: ComponentStatus) {
+        let mut components = match self.statuses.write() {
+            Ok(guard) => guard,
+            Err(poisoned) => poisoned.into_inner(),
```

Comment:
Any thoughts on migrating to the `parking_lot` crate to avoid dealing with lock poisoning?

Reply:
I think it makes sense. I initially was wary about adding extra dependencies to our codebase - which is incremental maintenance overhead. But it shouldn't affect compilation times since DataFusion already depends on it.

Reply:
Will deal with separately in https://github.com/spiceai/spiceai/issues/5499